Mage.Sets/src/mage/cards/r/RescueRetriever.java

```diff
@@ -0,0 +1,66 @@
+package mage.cards.r;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.common.PreventAllDamageToAllEffect;
+import mage.abilities.effects.common.counter.AddCountersAllEffect;
+import mage.constants.*;
+import mage.abilities.keyword.FlashAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.counters.CounterType;
+import mage.filter.FilterPermanent;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterArtifactCreaturePermanent;
+import mage.filter.common.FilterAttackingCreature;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+
+/**
+ *
+ * @author lunelis
+ */
+public final class RescueRetriever extends CardImpl {
+
+    private static final FilterCreaturePermanent filter = new FilterCreaturePermanent("other Soldier creature you control");
+    private static final FilterAttackingCreature filterAttacking = new FilterAttackingCreature("attacking soldiers you control");
+
+    static {
+        filter.add(SubType.SOLDIER.getPredicate());
+        filter.add(TargetController.YOU.getControllerPredicate());
```

Comment:
these two predicates don't need to be added here if you use the change I added above